.github/workflows/main.yaml

```
@@ -0,0 +1,42 @@
+name: Build and push redis package
+on:
+  push:
+    branches:
+      - main
+
+jobs:
+  build-and-push:
+    runs-on: ubuntu-latest
+    steps:     
+      - uses: actions/checkout@v4
+
+      - name: install spin
+        uses: fermyon/actions/spin/setup@v1
+        with:
+          github_token: ${{ github.token }}
+
+      - name: install go
+        uses: actions/setup-go@v5
+        with:
+          go-version: "1.22"
+
+      - name: install tinygo
+        uses: rajatjindal/setup-actions/tinygo@v0.0.1
+        with:
+          version: "v0.32.0"
+      
+      - name: build spinapp
+        run: spin build
+      
+      - name: login to GitHub container registry
+        uses: docker/login-action@v3
+        with:
+          registry: ghcr.io
+          username: ${{ github.actor }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: push package
+        run: |
+          RELEASE_VERSION=$(date +%Y%m%d-%H%M%S)-g$(git rev-parse --short HEAD) && \
+          spin registry push ghcr.io/${{ github.owner }}/${{ github.repository }}:$RELEASE_VERSION && \
+          spin registry push ghcr.io/${{ github.owner }}/${{ github.repository }}:latest
```

```suggestion
          spin registry push ghcr.io/${{ github.owner }}/${{ github.repository }}:$RELEASE_VERSION
```

I wouldn't bother publishing a `latest` image - it mostly results in documentation going stale quicker when you can't rely on pulling the same content.